File: src/dataset/cpp/isogram_Exercism_thefullarcticfox/openAI/ChatGPT4_isogram_v4.cpp
#include "isogram.h"
#include <unordered_set>
#include <cctype>

namespace isogram {
	bool is_isogram(const std::string& s) {
		std::unordered_set<char> seen;
		for (char c : s) {
			if (std::isalpha(c)) {
				char lower_c = std::tolower(c);
				if (!seen.insert(lower_c).second) {
					return false;
				}
			}
		}
		return true;
	}
}  // namespace isogram

File: src/dataset/typescript/isogram_Exercism-typescript-shybyte/openAI/ChatGPT4_isogram_v4.ts
export default class Isogram {
    static isIsogram(text: string): boolean {
        const seen = new Set();
        for (const char of text.toLowerCase()) {
            if (char !== ' ' && char !== '-' && seen.has(char)) return false;
            seen.add(char);
        }
        return true;
    }
}